client/src/components/VideoForm.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+
+function VideoForm({ videoData, setVideoData }) {
+  const [errorMessage, setErrorMessage] = useState("");
+
+  function addVideoHandler(event) {
+    /**
+     We are creating a new video object and adding to state 
+     */
+    function validateUrl(urlObject) {
+      console.log("urlObject --->", urlObject);
+      if (!urlObject.startsWith("https://www.youtube.com")) {
+        return false;
+      }
+      return true;
+      // return either validated URL or an error
+    }
+
+    event.preventDefault();
```

Comment:
Usually, **preventDefault()** is better to move to the top, just after the function addVideoHandler starts. This way, you minimise the potential bugs from rising. 

```
 function addVideoHandler(event) {
     event.preventDefault();
     // code here...
 }
```
It's worth having a read about what it does and why we need to invoke it. Here's the MDN [link](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault)

Reply:
This is very useful to know. Thank you!